Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1321,83 +1321,246 @@ auto RenderTheme::colorCache(OptionSet<StyleColorOptions> options) const -> Colo
     case CSSValueLinktext:
         return SRGBA<uint8_t> { 0, 0, 238 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-visitedtext
+    // Text in visited links. For light backgrounds, traditionally purple.
     case CSSValueVisitedtext:
         return SRGBA<uint8_t> { 85, 26, 139 };
-    case CSSValueActiveborder:
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-activetext
+    // Text in active links. For light backgrounds, traditionally red.
+    case CSSValueActivetext:
+    case CSSValueWebkitActivelink: // Non-standard addition.
+        return Color::red;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonface
+    // The face background color for push buttons.
+    case CSSValueButtonface:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttontext
+    // Text on push buttons.
+    case CSSValueButtontext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonborder
+    // The base border color for push buttons.
+    case CSSValueButtonborder:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-field
+    // Background of input fields.
+    case CSSValueField:
         return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-fieldtext
+    // Text in input fields.
+    case CSSValueFieldtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlight
+    // Background of selected text, for example from ::selection.
+    case CSSValueHighlight:
+        return SRGBA<uint8_t> { 181, 213, 255 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlighttext
+    // Text of selected text.
+    case CSSValueHighlighttext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-selecteditem
+    // Background of selected items, for example a selected checkbox.
+    case CSSValueSelecteditem:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-selecteditemtext
+    // Text of selected items.
+    case CSSValueSelecteditemtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-mark
+    // Background of text that has been specially marked (such as by the HTML mark element).
+    case CSSValueMark:
+        return Color::yellow;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-marktext
+    // Text that has been specially marked (such as by the HTML mark element).
+    case CSSValueMarktext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-graytext
+    // Disabled text. (Often, but not necessarily, gray.)
+    case CSSValueGraytext:
+        return Color::darkGray;
```

Comment:
Does this need to match the color of disabled text in system controls?

Reply:
I think using `RenderTheme::disabledTextColor` with black/white would be better here. It would match the disabled text color used in our form controls.

Reply:
This is existing, just shifted. I don't see a great reason to change it in this PR. 

As a side note, on macOS, CSSValueGraytext uses [NSColor disabledControlTextColor:] already.